### PR TITLE
aws-sdk-cpp 1.11.212

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-4163](https://anaconda.atlassian.net/browse/PKG-4163) 
- [Upstream repository](https://github.com/aws/aws-sdk-cpp/tree/1.11.212)
- Requirements:
  - https://github.com/aws/aws-sdk-cpp/tree/1.11.212?tab=readme-ov-file#building-the-sdk
  - https://github.com/aws/aws-sdk-cpp/blob/1.11.212/CMakeLists.txt
  - https://github.com/aws/aws-sdk-cpp/blob/1.11.212/docs/CMake_Parameters.md

### Explanation of changes:

- Enable echo on win
- Add `-DAWS_SDK_WARNINGS_ARE_ERRORS=OFF` from the conda-forge recipe
- Add `sqs;sns;monitoring;logs` to `-DBUILD_ONLY` (based on the conda-forge recipe)
- Build in parallel
- Remove macos settings from the local `cbc.yaml` because they are already in the base `cbc.yaml` 
- `zlib` dependency is new since the previous revision
- Use `ninja-base` in tests too

### Notes:

-

[PKG-4163]: https://anaconda.atlassian.net/browse/PKG-4163?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,8 +29,9 @@ requirements:
     - aws-c-event-stream 0.2.15
     - aws-crt-cpp 0.18.16
     - aws-checksums 0.1.13
-    - libcurl 7.88.1
+    - libcurl 8.9.1
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - libcurl {{ libcurl }}
```
This increases the range of curl versions installable with this package.

**Review by cbouss — APPROVED** [review]
LGTM, with the comment on libcurl addressed.

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: There are a few medium and one severe vulnerabilities in versions >=8.1.1,<8.9.1 https://curl.se/docs/vulnerabilities.html
Our current lubcurl is 8.1.1 https://github.com/AnacondaRecipes/aggregate/blob/master/conda_build_config.yaml#L134

Comment (cbouss) [reply recipe/meta.yaml]: Sure, but building with what is in the cbc will still resolve in the latest compatible on main.
In host: 8.1.1
In run: >=8.1.1,<9.0
Resolved when installed: 8.9.1

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Fixed

File: recipe/bld.bat [hunk 2]
```diff
@@ -7,7 +8,8 @@ cmake -LAH -G "Ninja" ^
       -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^
       -DCMAKE_INSTALL_LIBDIR=lib ^
       -DCMAKE_MODULE_PATH:PATH="%LIBRARY_LIB%\cmake" ^
-      -DBUILD_ONLY="s3;core;transfer;config;identity-management;sts" ^
+      -DAWS_SDK_WARNINGS_ARE_ERRORS=OFF ^
```
Comment (ifitchet) [inline recipe/bld.bat:11]: No warnings in `build.sh`?

**Review by ifitchet — APPROVED** [review]
Maybe a bit late but is it worth noting (in the PR's changes) that the `zlib` dependency is new since the previous revision?

**Pull request merged**